src/lib.rs

```diff
@@ -481,26 +482,65 @@ const INTERNAL_ERROR_MSG: &str = "Fatal internal error. Please consider filing a
                                   report at https://github.com/clap-rs/clap/issues";
 const INVALID_UTF8: &str = "unexpected invalid UTF-8 code point";
 
-/// @TODO @release @docs
-pub trait Clap: From<ArgMatches> + IntoApp + Sized {}
+/// This trait is just a convenience on top of FromArgMatches + IntoApp
+pub trait Clap: FromArgMatches + IntoApp + Sized {
```

Comment:
I'm convinced we shouldn't do that because:
* Some people may want to implement these traits manually, I've seen valid use cases for that.
* `clap_derive` docs are going to speak in terms of these traits, referencing them.
* `Foo doesn't implement IntoApp` errors won't make any sense for a user if we hide these traits .

Reply:
How about moving them into a mod called derive?

I just want to specify they are only there for derive.

Reply:
Will do